### DISCUSSION TITLE
A trivial observation for fellow time obsessives

### DISCUSSION THREAD

**Discussion by kjm1102** [micropython/micropython#12973] [General]
I've struggled a lot with timestamping events that occur in my esp32 code before the device knows what the current real time is. The micro starts counting from midnight Jan1 2000 so pre real-time-awareness timestamps have to be subsequently offset. The school of 'you did it wrong again stupid' has taught me to apply sensibility tests to anything I calculate in software. My de facto test for any timestamp, especially the offset ones, is to check it's epoch seconds is greater than the value at the time I wrote the code, crude but effective during code development in preventing wrongly stamped readings vanishing because the external database's house cleaning has erased them as they look to be too old. 

Anyway I was re-writing some code a couple of hours ago & instinctively checked the current epoch online to manually upgrade the utc value for
```
if Tstamp<utc-946684800: print('bogus epoch', Tstamp, 'is too old')
```
I sat there briefly mesmerised as the unix epoch rolled over past 1700000000 before my eyes, what are the odds?

**Comment by kjm1102** [comment]
I reckon I landed on that epoch converter site about 6 seconds before all
the zeros appeared.

**Reply by davefes** [reply]
The usual Kiwi-response is ... maybe you should buy a Lotto ticket :)

**Comment by kjm1102** [comment]
I should!
